state/execution.go

```diff
@@ -481,6 +481,15 @@ func fireEvents(
 		ResultEndBlock:   *abciResponses.EndBlock,
 	})
 
+	if len(block.Evidence.Evidence) != 0 {
+		for _, ev := range block.Evidence.Evidence {
+			eventBus.PublishEventNewEvidence(types.EventDataNewEvidence{
+				Evidence: ev,
+				Height:   block.Height,
```

Comment:
doesn't evidence always contain a height? `evidence.Height()`

Reply:
++

Reply:
evidence.Height() is the height at which the evidence happened, Height is the height that the evidence was committed on the chain. I thought having the height with which the evidence was committed might be useful if the person wants to query the entire block itself. As I said earlier it might not be that necessary so I'm indifferent if we want to remove it

Reply:
No, it might be useful.. let's leave it